### PR TITLE
Update compatibility docs and release cutting notes

### PR DESCRIPTION
**Pull request by sunjayBhatia**
- Adds release notes markdown template
- Calls out in release process doc about what to include in release
notes
- Modifies notes in compatibility matrix to remove ambiguity
- Minor updates in Philosophy document to bring it up to date
- Makes README more general about k8s prerequisites

### CODE REVIEW

File: site/content/resources/philosophy.md [hunk 1]
```diff
@@ -77,14 +77,13 @@ When designing Kubernetes objects, we try to expose information as close as poss
 For example, we will ensure that HTTPProxy objects have status conditions that tell the user that created them if there is a problem, rather than just logging that information from Contour itself.
 
 ### We meet users where they are
-Contour currently supports Ingress v1beta1, HTTPProxy and IngressRoute.
-In the near future, we’ll add support for Ingress v1 and after that Ingress v2.
+Contour currently supports Ingress v1, HTTPProxy and IngressRoute and we are working on supporting the Kubernetes Gateway API.
```
Comment (skriss) [inline site/content/resources/philosophy.md:80]:
```suggestion
Contour currently supports Ingress v1 and HTTPProxy and we are working on supporting the Kubernetes Gateway API.
```

File: site/content/resources/release-process.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ go run ./hack/release/prepare-release.go $CONTOUR_RELEASE_VERSION
 1. Document upgrade instructions for the new release (`site/content/resources/upgrading.md`).
 1. Commit all changes, push the branch, and PR it into `main`.
 
-_Note: the PR will probably fail the siteproof check due to [#2032](https://github.com/projectcontour/contour/issues/2032). It's a good idea to scan the CI log for any true issues._
+_Note: the PR will probably fail the siteproof check due to [#2032][3]. It's a good idea to scan the CI log for any true issues._
```
Comment (skriss) [inline site/content/resources/release-process.md:55]: we can drop this line now since we no longer have the siteproof check

**Review by skriss — CHANGES_REQUESTED** [review]
I think this looks good, just a couple nits. I'd suggest also updating https://github.com/projectcontour/contour#prerequisites to have more general language along the lines of "refer to the compatibility matrix for supported versions" so we don't have to remember to update it.

File: site/content/resources/philosophy.md (hunk 1 above)
Comment (skriss) [reply site/content/resources/philosophy.md:80]: no more IngressRoute :)

Comment (sunjayBhatia) [reply site/content/resources/philosophy.md]: good catch!

**Review by skriss — APPROVED** [review]
LGTM

**Comment by stevesloka** [comment]
Might need to also call out dropped support for Ingress v1beta1. I know it's removed from newer k8s versions, but could see users asking about this.

**Review by skriss — APPROVED** [review]

**Pull request merged**